binder_external_test.go

```diff
@@ -0,0 +1,130 @@
+// run tests as external package to get real feel for API
+package echo_test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/labstack/echo/v4"
+	"log"
+	"net/http"
+	"net/http/httptest"
+)
+
+func ExampleValueBinder_BindErrors() {
+	// example route function that binds query params to different destinations and returns all bind errors in one go
+	routeFunc := func(c echo.Context) error {
+		var opts struct {
+			Active bool
+			IDs    []int64
+		}
+		length := int64(50) // default length is 50
+
+		b := echo.QueryParamsBinder(c)
+
+		errs := b.Int64("length", &length).
+			Int64s("ids", &opts.IDs).
+			Bool("active", &opts.Active).
+			BindErrors() // returns all errors
+		if errs != nil {
+			for _, err := range errs {
+				bErr := err.(*echo.BindingError)
+				log.Printf("in case you want to access what field: %s values: %v failed", bErr.Field, bErr.Values)
+			}
+			return fmt.Errorf("%v fields failed to bind", len(errs))
+		}
+		fmt.Printf("active = %v, length = %v, ids = %v", opts.Active, length, opts.IDs)
+
+		return c.JSON(http.StatusOK, opts)
+	}
+
+	e := echo.New()
+	c := e.NewContext(
+		httptest.NewRequest(http.MethodGet, "/api/endpoint?active=true&length=25&ids=1&ids=2&ids=3", nil),
+		httptest.NewRecorder(),
+	)
+
+	_ = routeFunc(c)
+
+	// Output: active = true, length = 25, ids = [1 2 3]
+}
+
+func ExampleValueBinder_BindError() {
+	// example route function that binds query params to different destinations and stops binding on first bind error
+	failFastRouteFunc := func(c echo.Context) error {
+		var opts struct {
+			Active bool
+			IDs    []int64
+		}
+		length := int64(50) // default length is 50
+
+		// create binder that stops binding at first error
+		b := echo.QueryParamsBinder(c)
+
+		err := b.Int64("length", &length).
```

Comment:
It seems that this is not failing, as the name of the example states

Reply:
it is not meant to fail. It is just to show API for dealing single error. Examples have their own naming convention https://golang.org/pkg/testing/#hdr-Examples (go will complain if you are not using it) and this example point is to showcase `ValueBinder.BindError` method as with it you are dealing with single error ... usual go error check.